### PR TITLE
test: adds test content for orchestration provider

### PR DESCRIPTION
**Pull request by vilit1**
Looked at the base file and made some tests.
one of the shortest prs in a while
image
---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to Azure IoT Operations tooling!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

Intent for Production

- [ ] It is expected that pull requests made to default or core branches such as `dev` or `main` are of production grade. Corollary to this, any merged contributions to these branches may be deployed in a public release at any given time. By checking this box, you agree and commit to the expected production quality of code.

Basic expectations

- [ ] If introducing new functionality or modified behavior, are they backed by unit and/or integration tests?
- [ ] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [ ] Have **all** the relevant unit **and** integration tests pass? i.e. `pytest <project root> -vv`. Please provide evidence in the form of a screenshot showing a succesful run of tests locally OR a link to a test pipeline that has been run against the change-set.
- [ ] Have linter checks passed using the `.pylintrc` and `.flake8` rules? Look at the CI scripts for example usage.
- [ ] Have extraneous print or debug statements, commented out code-blocks or code-statements (if any) been removed from the surface area of changes?
- [ ] Have you made an entry in HISTORY.rst which concisely explains your user-facing feature or change?

Azure IoT Operations CLI maintainers reserve the right to enforce any of the outlined expectations.

A PR is considered **ready for review** when all basic expectations have been met (or do not apply).

### CODE REVIEW

File: azext_edge/tests/edge/init/test_base_unit.py [hunk 1]
```diff
@@ -0,0 +1,576 @@
...
+@pytest.mark.parametrize("mocked_resource_management_client", [{
+    "client_path": BASE_PATH,
+    "resources.begin_create_or_update_by_id": {"result": generate_generic_id()}
+}], indirect=True)
+@pytest.mark.parametrize("rotation_poll_interval", ["1h"])
+@pytest.mark.parametrize("extension_name", ["akvsecretsprovider"])
+def test_provision_akv_csi_driver(mocker, mocked_resource_management_client, rotation_poll_interval, extension_name):
+    terminal_result = mocker.Mock()
+    terminal_result.as_dict.return_value = generate_generic_id()
+    terminal_patch = mocker.patch(f"{BASE_PATH}.wait_for_terminal_state", autospec=True, return_value=terminal_result)
```
Comment (vilit1) [inline azext_edge/tests/edge/init/test_base_unit.py:28]: fixture

File: azext_edge/tests/edge/init/test_base_unit.py [hunk 2]
```diff
@@ -0,0 +1,576 @@
...
+            ]
+        }
+    }
+], ids=["pass everything"], indirect=True)
+@pytest.mark.parametrize("app_id", [None, generate_generic_id()])
+@pytest.mark.parametrize("object_id", [None, generate_generic_id()])
+@pytest.mark.parametrize("secret", [None, generate_generic_id()])
+@pytest.mark.parametrize("secret_valid_days", [365, 100])
+def test_prepare_sp(mocker, mocked_cmd, mocked_send_raw_request, app_id, object_id, secret, secret_valid_days):
+    tenant_patch = mocker.patch(f"{BASE_PATH}.get_tenant_id", return_value=generate_generic_id())
```
Comment (vilit1) [inline azext_edge/tests/edge/init/test_base_unit.py:174]: fixture

File: azext_edge/tests/edge/init/test_base_unit.py [hunk 3]
```diff
@@ -0,0 +1,576 @@
...
+    access_patch = mocker.patch(f"{BASE_PATH}.ensure_correct_access")
+    from azext_edge.edge.providers.orchestration.base import prepare_sp
+    deployment_name = generate_generic_id()
+    sp = prepare_sp(
+        mocked_cmd,
+        deployment_name,
+        service_principal_app_id=app_id,
+        service_principal_object_id=object_id,
+        service_principal_secret=secret,
+        service_principal_secret_valid_days=secret_valid_days
```
Comment (vilit1) [inline azext_edge/tests/edge/init/test_base_unit.py:184]: TIL; there are spies that could be used for secret_valid_days

File: azext_edge/tests/edge/init/test_base_unit.py [hunk 4]
```diff
@@ -0,0 +1,576 @@
...
+    assert params["properties"]["configurationProtectedSettings"] == {}
+    config_settings = params["properties"]["configurationSettings"]
+    assert config_settings["secrets-store-csi-driver.enableSecretRotation"] == enable_secret_rotation
+    assert config_settings["secrets-store-csi-driver.rotationPollInterval"] == rotation_poll_interval
+    assert config_settings["secrets-store-csi-driver.syncSecret.enabled"] == "false"
+
+
+@pytest.mark.parametrize("get_cluster", [False, True])
+def test_configure_cluster_secrets(mocker, get_cluster):
+    api_patch = mocker.patch(f"{BASE_PATH}.KEYVAULT_API_V1")
```
Comment (vilit1) [inline azext_edge/tests/edge/init/test_base_unit.py:67]: used in 2 places - could be fixture

File: azext_edge/tests/edge/init/test_base_unit.py [hunk 5]
```diff
@@ -0,0 +1,576 @@
...
+            sp_record=generate_generic_id()
+        )
+
+
+@pytest.mark.parametrize("get_cluster", [False, True])
+def test_configure_cluster_tls(mocker, get_cluster):
+    get_cluster_patch = mocker.patch(f"{BASE_PATH}.get_cluster_namespace", return_value=get_cluster)
+    create_cluster_patch = mocker.patch(f"{BASE_PATH}.create_cluster_namespace")
+    create_secret_patch = mocker.patch(f"{BASE_PATH}.create_namespaced_secret")
+    create_configmap_patch = mocker.patch(f"{BASE_PATH}.create_namespaced_configmap")
```
Comment (vilit1) [inline azext_edge/tests/edge/init/test_base_unit.py:132]: these funcs from ..base can be put into a fixture

File: azext_edge/edge/providers/orchestration/base.py [hunk 6]
```diff
@@ -331,6 +306,35 @@ def prepare_sp(cmd, deployment_name: str, **kwargs) -> ServicePrincipal:
     )
 
 
+def ensure_correct_access(cmd, sp_app_id: str, existing_resource_access: List[dict]):
```
Comment (digimaun) [inline azext_edge/edge/providers/orchestration/base.py:309]: cool refactor

**Review by digimaun — APPROVED** [review]

File: azext_edge/edge/providers/orchestration/base.py (hunk 6 above)
Comment (vilit1) [reply azext_edge/edge/providers/orchestration/base.py:309]: got tired of how big that freaking thing is

**Pull request merged**